transmogrifier/sources/transformer.py

```diff
@@ -229,67 +222,36 @@ def parse_source_file(cls, source_file: str) -> Iterator[dict[str, JSON] | Tag]:
         """
 
     @final
-    def _transform(
-        self, source_record: dict[str, JSON] | Tag
-    ) -> timdex.TimdexRecord | None:
+    def transform(self, source_record: dict[str, JSON] | Tag) -> timdex.TimdexRecord:
         """
-        Private method called for both XML and JSON transformations, where
-        all logic is shared except source_record type.
+        Transform source record into TimdexRecord instance.
+
+        Instantiates a TimdexRecord instance with required fields and runs fields methods
+        for optional fields. The optional field methods return values or exceptions that
+        prompt the __next__ method to skip the entire record.
 
         May not be overridden.
 
         Args:
             source_record: A single source record.
         """
         if self.record_is_deleted(source_record):
-            source_record_id = self.get_source_record_id(source_record)
-            timdex_record_id = self.get_timdex_record_id(
-                self.source, source_record_id, source_record
-            )
+            timdex_record_id = self.get_timdex_record_id(source_record)
             raise DeletedRecordEvent(timdex_record_id)
-        optional_fields = self.get_optional_fields(source_record)
-        if optional_fields is None:
-            return None
 
-        fields = {
-            **self.get_required_fields(source_record),
-            **optional_fields,
-        }
+        timdex_record = timdex.TimdexRecord(
+            source=self.source_name,
+            source_link=self.get_source_link(source_record),
+            timdex_record_id=self.get_timdex_record_id(source_record),
+            title=self.get_valid_title(source_record),
+        )
```

Comment:
Continue to like this init.  I find it comforting that if this block of explicit attempts for those required fields fail, this record will fail itself.